### PR TITLE
spiders: paraguay_dncp: request active/complete tender and planned ones separately

### LINKED ISSUE #882: `paraguay_dncp_*` get data by month
Currently, the pagination fails for pages larger than 1000. We can still get the data without errors if we always use enough filters, for example, requesting one month at a time.

### PR DESCRIPTION
**Pull request by yolile**
closes #882

### CODE REVIEW

File: kingfisher_scrapy/spiders/paraguay_dncp_base.py [hunk 1]
```diff
@@ -65,11 +65,16 @@ def urls_builder(self):
...
-                  f'fecha_desde={start_date.strftime(self.date_format)}-04:00&' \
-                  f'fecha_hasta={end_date.strftime(self.date_format)}-04:00&items_per_page=10000'
+            # We request active/complete tenders and planned ones separately to ensure we don't exceed the 10000
+            # results per request limit.
+            url_base = f'{self.base_url}/search/processes?fecha_desde={start_date.strftime(self.date_format)}-04:00' \
+                       f'&fecha_hasta={end_date.strftime(self.date_format)}-04:00&items_per_page=10000 '
+            # We request the active or successful tenders by using the "publicacion_llamado" filter.
+            url = f'{url_base}&tipo_fecha=publicacion_llamado'
+            # And the planned ones with the "fecha_release" and tender.id="planned" filters.
+            url_planning = f'{url_base}&tender.id="planned"&tipo_fecha=fecha_release'
```
Comment (jpmckinney) [inline kingfisher_scrapy/spiders/paraguay_dncp_base.py:75]: Don't `"` need to be escaped in URLs?

File: kingfisher_scrapy/spiders/paraguay_dncp_base.py [hunk 2]
```diff
@@ -65,11 +65,16 @@ def urls_builder(self):
...
                 start_date = end_date - interval
-            url = f'{self.base_url}/search/processes?tipo_fecha=fecha_release&' \
-                  f'fecha_desde={start_date.strftime(self.date_format)}-04:00&' \
-                  f'fecha_hasta={end_date.strftime(self.date_format)}-04:00&items_per_page=10000'
+            # We request active/complete tenders and planned ones separately to ensure we don't exceed the 10000
+            # results per request limit.
+            url_base = f'{self.base_url}/search/processes?fecha_desde={start_date.strftime(self.date_format)}-04:00' \
+                       f'&fecha_hasta={end_date.strftime(self.date_format)}-04:00&items_per_page=10000 '
+            # We request the active or successful tenders by using the "publicacion_llamado" filter.
+            url = f'{url_base}&tipo_fecha=publicacion_llamado'
```
Comment (jpmckinney) [inline kingfisher_scrapy/spiders/paraguay_dncp_base.py:73]: Should we name this `url_tender` to clarify the difference from `url_planning` ?

Comment (yolile) [reply kingfisher_scrapy/spiders/paraguay_dncp_base.py:73]: Sure, although that URL will contain planning information as well, but only the ones which have a tender already

File: kingfisher_scrapy/spiders/paraguay_dncp_base.py (hunk 1 above)
Comment (yolile) [reply kingfisher_scrapy/spiders/paraguay_dncp_base.py:75]: Actually, there is no need to use " at all

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**